provider/pkg/provider/deployment_settings.go

```diff
@@ -52,10 +57,47 @@ func (ds *PulumiServiceDeploymentSettingsInput) ToPropertyMap() resource.Propert
 				if ds.SourceContext.Git.GitAuth.SSHAuth != nil {
 					sshAuthPropertyMap := resource.PropertyMap{}
 					if ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey.Value != "" {
-						sshAuthPropertyMap["sshPrivateKey"] = resource.NewPropertyValue(FixMe)
+						if importMode {
+							importSecretValue(sshAuthPropertyMap, "sshPrivateKey", ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey, isInput)
+						} else {
+							if createMode {
+								createSecretValue(sshAuthPropertyMap, "sshPrivateKey", ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey,
+									plaintextInputSettings.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey, isInput)
+							} else {
+								if currentStateCipherSettings.SourceContext != nil &&
+									currentStateCipherSettings.SourceContext.Git != nil &&
+									currentStateCipherSettings.SourceContext.Git.GitAuth != nil &&
+									currentStateCipherSettings.SourceContext.Git.GitAuth.SSHAuth != nil {
+									mergeSecretValue(sshAuthPropertyMap, "sshPrivateKey", ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey,
+										plaintextInputSettings.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey,
+										currentStateCipherSettings.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey, isInput)
+								} else {
+									importSecretValue(sshAuthPropertyMap, "sshPrivateKey", ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey, isInput)
+								}
+							}
+						}
 					}
 					if ds.SourceContext.Git.GitAuth.SSHAuth.Password.Value != "" {
-						sshAuthPropertyMap["password"] = resource.NewPropertyValue(FixMe)
+						if importMode {
+							importSecretValue(sshAuthPropertyMap, "password", *ds.SourceContext.Git.GitAuth.SSHAuth.Password, isInput)
+						} else {
+							if createMode {
+								createSecretValue(sshAuthPropertyMap, "password", *ds.SourceContext.Git.GitAuth.SSHAuth.Password,
+									*plaintextInputSettings.SourceContext.Git.GitAuth.SSHAuth.Password, isInput)
+							} else {
+								if currentStateCipherSettings.SourceContext != nil &&
+									currentStateCipherSettings.SourceContext.Git != nil &&
+									currentStateCipherSettings.SourceContext.Git.GitAuth != nil &&
+									currentStateCipherSettings.SourceContext.Git.GitAuth.SSHAuth != nil &&
+									currentStateCipherSettings.SourceContext.Git.GitAuth.SSHAuth.Password != nil {
+									mergeSecretValue(sshAuthPropertyMap, "password", *ds.SourceContext.Git.GitAuth.SSHAuth.Password,
+										*plaintextInputSettings.SourceContext.Git.GitAuth.SSHAuth.Password,
+										*currentStateCipherSettings.SourceContext.Git.GitAuth.SSHAuth.Password, isInput)
+								} else {
+									importSecretValue(sshAuthPropertyMap, "password", *ds.SourceContext.Git.GitAuth.SSHAuth.Password, isInput)
+								}
+							}
+						}
```

Comment:
Can you explain why multiple paths result in `importSecretValue`? Why are we importing if we're not in import mode?

Reply:
For sure, and I will also add it as a comment
So if both current state and newly read state have some value, we normally are in merge mode (checking ciphertexts and deciding whether we need to clean plaintext value)
BUT
If new state retrieved has a new value, but old ciphertext does NOT have that value, that means we need to just import it

I was trying to rephrase that logic eloquently to avoid two branches importing, but it was even more ugly than now... I'll take one more stab at it

Reply:
> If new state retrieved has a new value, but old ciphertext does NOT have that value, that means we need to just import it

When does this happen?